java/client/src/main/java/glide/api/commands/servermodules/Json.java

```diff
@@ -707,6 +709,149 @@ public static CompletableFuture<Long> arrlen(
         return executeCommand(client, new GlideString[] {gs(JSON_ARRLEN), key});
     }
 
+    /**
+     * Reports debug information of the given <code>type</code> of the object at the specified <code>
+     * path</code> within the JSON document stored at <code>key</code>.
+     *
+     * @param client The client to execute the command.
+     * @param type The debug information type, see {@link JsonDebugType}.
+     * @param key The key of the JSON document.
+     * @param path The path within the JSON document.
+     * @return
+     *     <ul>
+     *       <li>For JSONPath (<code>path</code> starts with <code>$</code>):<br>
+     *           Returns an <code>Object[]</code> with a list of numbers for every possible path,
+     *           indicating the debug information requested, or <code>null</code> for JSON values
+     *           matching the path that are not an array. If <code>path</code> does not exist, an
+     *           empty array will be returned.
+     *       <li>For legacy path (<code>path</code> doesn't start with <code>$</code>):<br>
+     *           Returns an integer representing the debug information requested. If multiple paths
+     *           are matched, returns the data of the first matching object. If <code>path</code>
+     *           doesn't exist or the value at <code>path</code> is not an array, an error is raised.
+     *     </ul>
+     *     If <code>key</code> doesn't exist, returns <code>null</code>.
+     * @example
+     *     <pre>{@code
+     * Json.set(client, "doc", "$", "[1, 2.3, \"foo\", true, null, {}, [], {\"a\":1, \"b\":2}, [1, 2, 3]]").get();
+     * var res = Json.debug(client, FIELDS, "doc", "$[*]").get();
+     * assert Arrays.equals((Object[]) res, new Object[] {1, 1, 1, 1, 1, 0, 0, 2, 3});
+     *
+     * res = Json.debug(client, MEMORY, "doc", "..").get();
+     * assert res == 258L;
+     * }</pre>
+     */
+    public static CompletableFuture<Object> debug(
```

Comment:
do want to follow the python version api?  where they provide separate api's for each command?  
https://github.com/valkey-io/valkey-glide/pull/2481

Reply:
Redis JSON kind of splits the sub commands out separately. e.g. https://redis.io/docs/latest/commands/json.debug-memory/